src/lib/atoms/Rating/Rating.tsx

```diff
@@ -0,0 +1,268 @@
+        setRegardingPosition(0);
+        setRating(temporaryRatingValue);
+    };
+
+    const getRating = (currentRating: number) => {
+        if (readonly) return;
+
+        const state = regardingPosition === 50 ? +`${currentRating - 1}.${regardingPosition}` : currentRating;
+
+        const getCurrentRateValue = (prev: number) => {
+            if (state !== prev) return state;
+            return 0;
+        };
+
+        if (onChange) {
+            onChange(state);
+        }
+        setTemporaryRatingValue(getCurrentRateValue);
+        setRating(getCurrentRateValue);
+    };
+
+    const getRef = (el) => {
+        if (el) {
+            iconRef.current = el;
+            return;
+        }
+    };
+
+    const mouseEnterHandler = () => {
+        setTemporaryRatingValue(rating);
+    };
+
+    const iconsToRender = count > 0 ? new Array(count).fill(undefined) : [];
+    const currentSize = size === 'small' ? '32px' : size === 'big' ? '42px' : '36px';
+    const generalPropsForIcons = {
+        width: '100%',
+        height: currentSize
+    };
+
+    const DefaultElement = ({ isIcon = false, Element }) =>
+        isIcon ? (
+            <Element
+                fill={color}
+                stroke={'0'}
+                strokeWidth={0}
+                ref={getRef}
+                {...generalPropsForIcons}
+                className={classNames('rating__icon', `s-${size}`)}
+            />
+        ) : (
+            <div
+                style={{
+                    color
+                }}
+                ref={getRef}
+                {...generalPropsForIcons}
+                className={classNames('rating__icon', `s-${size}`)}
+            >
+                {Element}
+            </div>
+        );
+
+    const CompareElement = ({ isIcon = false, Element }: { isIcon?: boolean; Element: ReactNode | FC<any> }) => {
+        const Icon = isIcon && !isValidElement(Element) ? (Element as FC<SVGProps<'classNames'>>) : null;
+        return Icon ? (
+            <Icon
+                {...generalPropsForIcons}
+                fill={backgroundColor}
+                className={classNames('rating__icon', `s-${size}`)}
+                stroke={'0'}
+                strokeWidth={'0'}
+            />
+        ) : (
+            <div
+                style={{
+                    color: backgroundColor
+                }}
+                {...generalPropsForIcons}
+                className={classNames('rating__icon', `s-${size}`)}
+            >
+                {Element as ReactNode}
+            </div>
+        );
+    };
+
+    return (
+        <div {...restProps} className="rating" onMouseLeave={mouseLeave} onMouseEnter={mouseEnterHandler}>
+            {iconsToRender.map((_, i) => {
+                const currentRating = i + 1;
+
+                const Callback =
+                    typeof character === 'function' &&
+                    isValidElement(character(i)) &&
+                    (character(i) as JSX.Element).type
```

Comment:
Let's change `JSX.Element` to `ReactElement` since the `JSX.Element` is deprecated.

```suggestion
                    (character(i) as ReactElement).type
```